src/Workspaces/Core/Portable/Workspace/Solution/TextDocumentState.cs

```diff
@@ -143,15 +150,20 @@ public TextDocumentState UpdateText(SourceText newText, PreservationMode mode)
         return UpdateText(newTextAndVersion, mode);
     }
 
-    public TextDocumentState UpdateText(TextLoader loader, PreservationMode mode)
+    public TextDocumentState UpdateText(TextLoader? loader, PreservationMode mode)
```

Comment:
what does it mean for the loader to go to null.  Allowing nulls to flow into these places gives me the heebie jeebies.

Reply:
Have to allow null here since `DocumentInfo.TextLoader` is nullable public property:

```
/// <summary>
/// A loader that can retrieve the document text.
/// </summary>
public TextLoader? TextLoader { get; }
```

Reply:
Why is there no crying emoji...

What does that even mean...  bleagh.

My preference would be that that would be non-null.  And then at a sole entrypoint where we read this, we override a null loader with one that returns an empty string.  Having the loader itself be null, and passed through the system is just so confusing.

no neeed to change this now though if that's what our API is.  bleagh.